packages/cli/internal/pkg/cli/context_deploy.go

```diff
@@ -89,45 +84,94 @@ func (o *deployContextOpts) validateSuppliedContexts(contextList []string) error
+		if progressResult.Err != nil {
+			failedDeployments = append(failedDeployments, progressResult)
+		}
+	}
+
+	failedDeploymentsLength := len(failedDeployments)
+	if failedDeploymentsLength > 0 {
+		for i, failedDeployment := range failedDeployments {
+			log.Error().Msgf("Failed to deploy context '%s'. Below is the log for that deployment", failedDeployment.Context)
+
+			outputsLength := len(failedDeployment.Outputs)
+			for i := 0; i < outputsLength-1; i++ {
+				log.Error().Msg(failedDeployment.Outputs[i])
+			}
+
+			if i < failedDeploymentsLength-1 {
+				log.Error().Msgf("%s \n\n\n", failedDeployment.Outputs[outputsLength-1])
+			} else {
+				log.Error().Msg(failedDeployment.Outputs[outputsLength-1])
+			}
+
 			var actionableError *actionableerror.Error
-			ok := errors.As(result.err, &actionableError)
+			ok := errors.As(failedDeployment.Err, &actionableError)
 			if ok {
 				log.Error().Err(actionableError.Cause).Msgf(actionableError.Error())
 				aggregateSuggestions = append(aggregateSuggestions, actionableError.SuggestedAction)
-			} else {
-				log.Error().Err(result.err).Msgf("failed to deploy context '%s'", result.contextName)
 			}
-			hasErrors = true
 		}
-		contextDetails[i] = result.info
-	}
-	if hasErrors {
-		aggregateSuggestions = slices.DeDuplicateStrings(aggregateSuggestions)
-		return nil, actionableerror.New(fmt.Errorf("one or more contexts failed to deploy"), strings.Join(aggregateSuggestions, ", "))
+
+		return actionableerror.New(fmt.Errorf("one or more contexts failed to deploy"), strings.Join(aggregateSuggestions, ", "))
 	}
-	sortContextDetails(contextDetails)
-	return contextDetails, nil
+
+	return nil
 }
 
-func (o *deployContextOpts) deployContexts(contexts []string) []deployResult {
-	results := make([]deployResult, len(contexts))
-	for i, contextName := range contexts {
-		log.Debug().Msgf("Deploying context '%s'", contextName)
-		// TODO: Run in parallel once CDK resolves race condition causing context bleed
-		//       https://github.com/aws/aws-cdk/issues/14350
-		func(ctxManager context.Interface, i int, contextName string) {
-			_ = ctxManager.Deploy(contextName, true)
+func (o *deployContextOpts) validateDeploymentResults() ([]context.Detail, error) {
+	contextDetails, deploymentHasErrors := make([]context.Detail, len(o.contexts)), false
+	aggregateSuggestions := make([]string, 0)
+	var wait sync.WaitGroup
+	wait.Add(len(o.contexts))
+
+	for i, contextName := range o.contexts {
+		go func(ctxManager context.Interface, i int, contextName string) {
 			info, err := ctxManager.Info(contextName)
-			results[i] = deployResult{contextName: contextName, info: info, err: err}
+			contextDetails[i] = info
+			if err != nil {
+				var actionableError *actionableerror.Error
+				ok := errors.As(err, &actionableError)
+				if ok {
+					log.Error().Err(actionableError.Cause).Msgf(actionableError.Error())
+					aggregateSuggestions = append(aggregateSuggestions, actionableError.SuggestedAction)
```

Comment:
I'm trying to think of when an actionable error from Info would ever be useful here for Deploy but I'm failing to come up with any

Reply:
I've experienced the invalid credential when deploying contexts after they succeeded. It's also safer incase we do actually pass through an actionable error.